### ISSUE TITLE
`too many open files, scandir 'xxx'` error in directory with many files in the lower level

### ISSUE THREAD

**Issue by dqn** [uhyo/nitrogql#10]
When I run `npx nitrogql check`, it appears to be reading directories unrelated to the config and giving an error.

Here is the full text of the error (some names have been changed):
```
$ npx nitrogql check
node:fs:1456
  handleErrorFromBinding(ctx);
  ^

Error: EMFILE: too many open files, scandir '/path/to/my-project/lower/level/directory'
    at Object.readdirSync (node:fs:1456:3)
    at fd_readdir (/path/to/my-project/node_modules/@nitrogql/cli/bin/shim.mjs:78:16)
    at wasm://wasm/00718a4e:wasm-function[342]:0x134b88
    at wasm://wasm/00718a4e:wasm-function[67]:0xac75f
    at wasm://wasm/00718a4e:wasm-function[121]:0xde0c8
    at wasm://wasm/00718a4e:wasm-function[51]:0x9579b
    at wasm://wasm/00718a4e:wasm-function[22]:0x1b20f
    at wasm://wasm/00718a4e:wasm-function[1500]:0x19debf
    at wasm://wasm/00718a4e:wasm-function[495]:0x157933
    at wasm://wasm/00718a4e:wasm-function[725]:0x1776ae {
  errno: -24,
  syscall: 'scandir',
  code: 'EMFILE',
  path: '/path/to/my-project/lower/level/directory'
}

Node.js v19.8.1
```

## Environment

```
Operating System:
  Platform: darwin
  Arch: x64
  Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:42:11 PST 2023; root:xnu-8792.81.3~2/RELEASE_X86_64
Binaries:
  Node: 19.8.1
  npm: 9.5.1
```

**Comment by dqn** [comment]
I noticed that in 1.0.0-beta.2 the `npx nitrogql check` does not complete in my environment, regardless of whether there are many or few files (directories?). In 1.0.0-beta.0 or 1.0.0-beta.1, it completes immediately if there are few directories.

**Comment by dqn** [comment]
I have found that placing just four files in a particular directory structure causes `too many open files 'xxx'` error. If one of these files is missing, the error will no longer occur lol. I am researching the detailed conditions.

**Comment by dqn** [comment]
I have prepared perhaps the smallest pattern to reproduce this (v1.0.0-beta.0). On macOS, the above error occurs. On Windows, the command does not complete.
https://github.com/dqn/nitrogql-bug-reproduction-20230411

This may not reproduce if the file or directory name is changed. I don't know why the error occurs in this case, but in any case it may need to be fixed so that it does not read directories that are not unrelated to the config.

**Comment by uhyo** [comment]
Memo: on my M1 Mac, the command does not complete. 🥲

**Comment by uhyo** [comment]
I released v1.0.0-beta.3. The problem is fixed on my side with this version. I hope you can try. :pray:

(By the way, this version should work but it still looks into irrelevant directories. To fix this behavior I have to create my own glob implementation. )
